### PR TITLE
cli: make initializing the global meter- and tracing providers optional

### PR DESCRIPTION
**Pull request by thaJeztah**
- relates to https://github.com/docker/cli/commit/02537eac596a6e1bab435c8d2445603990445678 / https://github.com/docker/cli/pull/5067

**- Description for the changelog**

```markdown changelog
Add WithEnableGlobalMeterProvider and WithEnableGlobalTracerProvider options
```

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by laurazard — APPROVED** [review]
LGTM

File: cli/command/cli.go [hunk 1]
```diff
@@ -92,6 +92,8 @@ type DockerCli struct {
 	// this may be replaced by explicitly passing a context to functions that
 	// need it.
 	baseCtx context.Context
+
+	enableGlobalMeter, enableGlobalTracer bool
```
Comment (krissetto) [inline cli/command/cli.go:96]: nit: we should probably call these Meter/Tracer**Provider**

**Review by krissetto — APPROVED** [review]
Tiny nit but LGTM

File: cli/command/cli.go (hunk 1 above)
Comment (thaJeztah) [reply cli/command/cli.go:96]: Yeah, they were already far too  long names for an un-exported field 😂

File: cli/cobra.go [hunk 2]
```diff
@@ -166,6 +166,7 @@ func (tcmd *TopLevelCommand) HandleGlobalFlags() (*cobra.Command, []string, erro
 	// arguments are in `flags.Args()`.
 	if err := flags.Parse(tcmd.args); err != nil {
 		// Our FlagErrorFunc uses the cli, make sure it is initialized
+		// TODO(thaJeztah): does this need OTEL global tracer and meter to be configured?
```
Comment (thaJeztah) [inline cli/cobra.go:169]: Oh! Can someone have a quick peek here? Do I need to keep this TODO or safe to remove?

Comment (krissetto) [reply cli/cobra.go:169]: the global providers _shouldn't_ be necessary here

Comment (thaJeztah) [reply cli/cobra.go]: Thx! Removed

**Review by vvoland — APPROVED** [review]

**Pull request merged**